src/eval/mod.rs

```diff
@@ -1640,17 +1688,28 @@ impl Eval for ast::ModuleInclude {
     fn eval(&self, vm: &mut Vm) -> SourceResult<Self::Output> {
         let span = self.source().span();
         let source = self.source().eval(vm)?;
-        let module = import(vm, source, span)?;
+        let module = import(vm, source, span, false)?;
         Ok(module.content())
     }
 }
 
 /// Process an import of a module relative to the current location.
-fn import(vm: &mut Vm, source: Value, span: Span) -> SourceResult<Module> {
+fn import(
+    vm: &mut Vm,
+    source: Value,
+    span: Span,
+    accept_functions: bool,
+) -> SourceResult<Module> {
     let path = match source {
         Value::Str(path) => path,
         Value::Module(module) => return Ok(module),
-        v => bail!(span, "expected path or module, found {}", v.type_name()),
+        v => {
```

Comment:
could it maybe be easier to just dump a whole `Module` into the function instead of a `Scope` and handle everything here? or create a module here on demand.

Reply:
I'm thinking that a `Module` would be excessive, as it also contains a name (which the function already has) and some content field (I didn't quite get what that is for, but anyway, lol). Maybe we should keep this function exclusive to just handling paths (since, for modules, it just returns them straight away), and do the matching outside this function (since it's different for import and include - the former also accepts functions)?

Reply:
The content is for `#include`. Markup files also become modules.